Refeito o frontend em svelte/sapper

GCastilho opened the pull request:
-- Refeito o frontend em svelte, usando o sapper como roteador
-- Todas as funcionalidades já implementadas no ReactJS foram implementadas no svelte
-- Removido o ReactJS completamente do projeto

Geral:
-- Removido arquivo obsoleto .npmrc
-- Corrigido um bug no script build:clean
-- Adicionado scripts 'dep-install' para instalar dependências de desenvolvimento em todos os sub-repositórios do projeto

Servidor:
-- O método 'authentication' foi separado em 'authenticate' e 'deauthenticate', ambos agora retornam um callback com o sucesso/erro da conexão
-- Removido middleware de autenticação na conexão, agora a autenticação é feita apenas usando os eventos 'authenticate' e 'deauthenticate'
-- Refatorado o roteador do websocket para funcionar sem um path na conexão e no re-roteamento (presume home)
-- Os listeners globais foram removidos do index do websocket e colocados em um arquivo 'globalListeners'
-- O evento 'list' da balances foi adaptado para retornar o callback no padrão 'err, response'
-- O evento 'list' da balances não mais retorna o saldo do cliente
-- A getBalance da User agora retorna um objeto com os saldos 'available' e 'locked' ao invés do valor do 'available'
-- Foi adicionado um evento global 'fetch_balances' que retorna um objeto em que as chaves são os nomes das currencies e os valores são os saldos do usuário para essa currency (type number)

Svelte:
-- Implementada uma função 'route' no websocket para fazer o roteamento do socket usando o evento '_path'
-- Implementado uma função 'emit' no websocket para ser um wrapper em torno do callback do mesmo, a promessa que ele retorna só será resolvida caso o socket esteja conectado ou uma vez que ele se conectar (o evento só será emitido no socket caso ele esteja conectado, não dando um erro de 'SocketDisconnected')
-- O websocket não mais envia um header 'path' ao iniciar a conexão
-- Implementado uma store de autenticação com o socket com métodos para autenticar e desautenticar o socket; Seu valor é o estado atual da autenticação (true ou false)
-- Implementado uma store para armazenar o saldo client-side, o valor da store será o objeto retornado pelo fetch_balance
-- O token de autenticação com o socket (atualmente o sessionID) agora é salvo no localStorage
-- Ao conectar-se, o módulo da store de autenticação irá tentar se autenticar com o servidor automaticamente usando o token salvo no localStorage (caso exista)

GCastilho committed:
GREAT SUCCESS!

-- Sapper send GREAT SUCCESS!
-- Implementado um frontend em sapper/svelte em '/sapper' usando o template padrão do sapper
-- Configurado servidor de dev para redirecionar todos os requests não GET para o backend
-- A porta padrão do backend agora é a 3000
-- Os scripts npm de dev agora setam a porta do backend para 3001 antes de iniciarem

GCastilho committed:
Consertado eslint no sapper

-- Adcionado um .eslintrc.json no /sapper para ter as configurações do linter personalizadas para o frontend
-- Configurado o linter do sapper para reconhecer módulos ES6
-- Feito um linter automatizado para remover configs incorretas (como extra ;)
-- Removido arquivo obsoleto .npmrc

GCastilho committed:
Adicionado scripts de compilação

-- Adicionado um script (temporário) build-svelte que faz a mesma coisa que o build mas com o svelte; Eventualmente substituirá o build
-- Corrigido um bug no script build:clean
-- Adicionado scripts para instalar dependências de desenvolvimento em todos os sub-repositórios do projeto
-- O script 'export' do sapper agora irá mover o repositório exportado para a pasta '../public'
-- Corrigido um bug no proxy do modo dev em que ele não estava redirecionando requests do websocket corretamente

joaojvds reviewed (dismissed):
Para os próximos arquivos é bom separar o css em outros arquivos, porque se não ele vai ficar ocupando a maior parte do código

GCastilho commented:
@joaojvds Pq mesmo você aprovou um PR que ainda está em draft?

joaojvds commented:
@GCastilho  Não vi que era um draft

GCastilho committed:
Feito o design da página de cadastro

-- Adicionado um menu na navBar para as páginas de login e de cadastro
-- Feito o design da página de cadastro, completamente diferente do que era em react
-- Dois compenentes auxiliares foram criados, um "fancyButton" e um "fancyInput", que são genéricos e podem ser utilizados em outros lugares do código

GCastilho committed:
Implementado handler para o formulário de login

-- Implementado a funcionalidade do formulário de cadastro, ele agora envia o request POST ao servidor e processa a resposta
-- Adicionado um box de mensagem de erro (igual do react) para informar o usuário de erros no request de cadastro (funcionalidade idêntica a sua versão em react)
-- Implementado a mensagem de confirmação de email

GCastilho committed:
Implementado a página de login

-- Implementado a página de login, com o mesmo estilo da página de cadastro
-- Implementado a funcionalidade da página de login, como processamento de erros e o redirect para a URL informada pelo servidor

GCastilho committed:
Implementado o websocket

-- Corrigido um bug no redirecionamento ao fazer login, ele agora é hardcoded para a home
-- Corrigido um bug no servidor dev do sapper que não estava redirecionando os sockets nossos para o servidor
-- Adicionado um módulo 'websocket' para conter funções relacionada ao websocket
-- O módulo expõe uma função 'route' que envia o evento '_path' com o argumento que ela recebe
-- Foi adicionado na navbar o listener para mudança de rota que chama a função Route do socket
-- O módulo do websocket também expõe uma função 'emit' para emitir eventos no socket e duas outras funções para adicionar e remover listeners dele

GCastilho committed:
Websocket não usa mais path no header

-- O weboscket não mais envia um header 'path' ao iniciar a conexão. Isso faz o path '/' ser presumido pelo server, o que não é um problema pq um evento '_path' é sempre enviado quando a página carrega (O server deverá ser refatorado para remover código inútil)
-- Refatorado o proxy do modo dev para redirecionar corretamente o socket agora que não existe mais um campo path
-- Refatorado o código da Nav que roteira o socket e deixado mais simples, já que não é mais necessário importar o websocket no onMount

GCastilho committed:
Implementado uma store de autenticação

-- Implementado uma store de autenticação com o socket em 'stores/auth.js', essa store tem matodos para autenticar e desautenticar o socket e seu valor é o estado atual da autenticação (true ou false)
-- A página de login agora chama o método 'authenticate' da store de autenticação para autenticar a conexão com o socket assim que o usuário se autentica no sistema
-- Adicionado um botão de logout na navbar que aparece apenas qdo o usuário está logado
-- Tentar acessar as páginas de login ou cadastro ao estar logado fará o cliente ser redirecionado para a home

GCastilho committed:
Adicionado uma página de balances

-- Adicionado uma página de balances para mostrar o saldo do usuário nas currencies do sistema
-- Criado o design base da tabela de balances
-- Resolvido um bug no server do sapper que não estava redirecionando requests ao main server corretamente

GCastilho committed:
Os botões no menu de Actions agora abrem e fecham as abas

-- O arquivo balances.svelte agora é index.svelte na pasta balances
-- Separado a linha da tabela em um componente separado _tableRow
-- Adicionado funções para abrir e fechar o menu de ações ao clicar em 'Deposit' e 'withdraw'

GCastilho committed:
Implementado menu de depósito

-- Implementado um menu de depósito na TableRow, com suporte a um cliente ter múltiplos endereços
-- O primeiro endereço do array de address retornado será selecionado automaticamente sempre
-- Implementado um QR Code que mostra a URI name:address da account selecionada

GCastilho committed:
Implementado o menu de withdraw

-- Implementado um menu de withdraw para fazer saques no sistema
-- Implementado funcionalidade para impedir o usuário de digitar um saldo maior que o disponível para saque
-- Ao executar um saque bem sucedido, o saldo será atualizado usando o valor do amount

GCastilho committed:
Refatorado TableRow

-- Refatorado TableRow e colocado os componentes elementos da actionCell em seus próprios componentes
-- A TableRow e seus componentes foram colocados em uma pasta _TableRow por questões de hierarquia

GCastilho committed:
Removido ReactJs do projeto

-- Com a implementação do Svelte em par com o que o React tinha, o React foi completamente removido do projeto
-- Atualizado scripts de compilação e dev para utilizarem o svelte; Removido scripts temporários
-- Atualizado dependências do cliente

GCastilho committed:
Adicionado systema de autenticação persistente

-- O token de autenticação com o socket (atualmente o sessionID) agora é salvo no localStorage, e não na memória, permitindo autenticação persistente no site
-- A página de balances agora necessita de o usuário estar autenticado para ser aberta, o usuário será redirecionado para a página de login em caso negativo
-- Ao abrir o site, se o usuário tiver um token salvo o sistema irá presumir que ele está autenticado; Isso é feito assim para que ele não seja redirecionado de páginas que necessitam autenticação (como a balance) pois o socket não conectou e autenticou antes de a página terminar de carregar (tipo sempre)

GCastilho committed:
Refeito o sistema de autenticação com o websocket

-- Removido a capacidade do websocket de autenticar no momento da conexão
-- O método 'authentication' foi separado em 'authenticate' e 'desauthenticate', ambos agora retornam um callback com o sucesso/erro da conexão
-- Refatorado o sistema de autenticação do cliente para não mais presumir que a autenticação/desautenticação foi bem-sucedida, já que o servidor agora retorna um callback com o status
-- Ao conectar-se, o cliente irá tentar se autenticar com o servidor automaticamente usando o token salvo no localStorage (caso exista)

GCastilho committed:
List agora é uma função global

-- Refatorado o roteador do websocket para funcionar sem um path na conexão e no re-roteamento (presume home)
-- Os listeners globais foram removidos do index do websocket e colocados em um arquivo 'globalListeners'
-- O evento 'list' agora é global e foi movido da balances para a globalListeners; Seu callback foi arrumado para seguir o padrão 'err, response'

GCastilho committed:
Consertado uma implementação incorreta do listener de conexão

-- Consertado uma implementação incorreta do listener do auth pelo evento 'connect' do socket, que estava usando o socket diretamente ao invés da addEventListener
-- Consertado o websocket estar exportando o socket diretamente
-- Refatorado a implementação do 'auth' para manter todos os acessos ao localStorage em 3 funções helpers
-- Refatorado como o "presume autenticado se existe um token" funciona, ele agora inicializa a store com true/false ao invés de setar ela logo depois de inicializar

GCastilho committed:
Separado evento 'list' e 'fetch_balance' no websocket

-- O evento 'list' do websocket voltou para a balances, mas agora ele não retorna o balance em sí, só o resto
-- A getBalance da User agora retorna um objeto com os saldos 'available' e 'locked' ao invés do valor do 'available'
-- Foi adicionado um evento global 'fetch_balances' que retorna um objeto em que as chaves são os nomes das currencies e os valores são os saldos do usuário (type number)

GCastilho committed:
Adicionado uma store para armazenar o saldo client-side

-- Adicionado uma store para armazenar o saldo client-side, o valor da store será o objeto retornado pelo fetch_balance
-- Adaptado a página de balance para uutilizar essa store para mostrar e manupular os dados do saldo
-- Refatorado os botões de deposit e withdraw para terem um funcionamento mais claro
-- Corrigido um bug no initialState do auth estar sendo setado como string e não boolean

GCastilho committed:
Melhorado o funcionamento do websocket

-- A promessa retornada pela função emit do websocket agora só irá resolver caso o socket esteja conectado ou uma vez que ele se conectar, ou seja, o evento só será emitido no socket caso ele esteja conectado
-- A função emit não rejeta mais um erro "SocketDisconnected"
-- Refatorado a store de blances para o novo funcionamento do websocket

GCastilho committed:
Cleanup

-- Feito limpeza, reorganização e leve refatoramento no código
-- Removido arquivo README do cliente
-- Atualizado dependências do cliente
-- Removido seção "blog" do site
-- Removido função temporária 'list' da página de balances e substituído por uma call direta ao socket
-- Refatorado o redirecionamento de usuário da página de balance caso ele não esteja autenticado
-- Minor fixes

GCastilho committed:
Resolvido bug no formulário de saque

-- Resolvido bug no formulário de saque em que a filterAmount estava usando uma variável local (inexistente) ao invés da store de saldo e causando uma exception

GCastilho committed:
Resolvido bug em que a URIdo QRCode não estava sendo sendo mostrada corretamente

-- Resolvido um bug onde a URI do QRCode ficava :address e não currency:address

joaojvds reviewed (approved):
Nenhum outro erro encontrado
